Messages/MessageDecorator.cs

```diff
@@ -0,0 +1,19 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Messages
+{
+    public class MessageDecorator : Message
+    {
+        public Message DecoratedMessage { get; set; }
+        public override int MsgId => DecoratedMessage.MsgId;
```

Comment:
Can you explain to me what this does?

Reply:
So if it has a Message as a member variable why does it also hold the member ID as seperate variables?

Reply:
Because if it didn't, when you do:
`msg = new CreateGameDecorator(){...}` It then no longer has a `MsgID` or `ConvID` because when `new` is called and its setting it as a new class, it creates a completely new instance. With `MessageDecorator` holding and overriding an instance of `MsgID` and `ConvID`, its keeping them to be used with the decorating classes.